### PR TITLE
H5 logging

### PR DESCRIPTION
**Pull request by wuallenwu**

### CODE REVIEW

**Comment by wuallenwu** [comment]
also added some optimizations to stompy mini - it trains pretty fast (100k+ steps per second) with collisions off, but much slower with collisions off

an even lighter version of stompy mini is here https://drive.google.com/drive/folders/1fziT-K6y6BhRszKmsn5YLawPwj-oJgON?usp=drive_link

File: sim/humanoid_gym/play_h5.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    # Prepare for logging
+    env_logger = Logger(env.dt)
+    robot_index = 0
+    joint_index = 1
+    stop_state_log = 1000
+
+    # Initialize HDF5 file
+    now = datetime.now()
+    now = now.strftime("%Y-%m-%d_%H-%M-%S")
+    h5_file = h5py.File("data" + now + ".h5", "w")
```
Comment (budzianowski) [inline sim/humanoid_gym/play_h5.py:79]: @WT-MM - can you have a look

File: sim/humanoid_gym/envs/stompy_mini_env.py [hunk 2]
```diff
@@ -265,6 +265,7 @@ def compute_observations(self):
 
         self.obs_buf = obs_buf_all.reshape(self.num_envs, -1)  # N, T*K
         self.privileged_obs_buf = torch.cat([self.critic_history[i] for i in range(self.cfg.env.c_frame_stack)], dim=1)
+        # breakpoint()
```
Comment (budzianowski) [inline sim/humanoid_gym/envs/stompy_mini_env.py:268]: remove

File: sim/humanoid_gym/play_h5.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        # Store h5 data
+        for i in range(buf_len):
+            cur_obs = env.obs_history[i].tolist()[0]
+            dset_2D_command[t, i] = cur_obs[0:2]
+            dset_3D_command[t, i] = cur_obs[2:5]
+            dset_q[t, i] = cur_obs[5 : 5 + num_dof]
+            dset_dq[t, i] = cur_obs[5 + num_dof : 5 + 2 * num_dof]
+            dset_obs_actions[t, i] = cur_obs[5 + 2 * num_dof : 5 + 3 * num_dof]
+            dset_ang_vel[t, i] = cur_obs[5 + 3 * num_dof : 8 + 3 * num_dof]
+            dset_euler[t, i] = cur_obs[8 + 3 * num_dof : 11 + 3 * num_dof]
```
Comment (WT-MM) [inline sim/humanoid_gym/play_h5.py:171]: Add comment describing obs structure e.g. 2d command (2), 3d command (3), etc.

File: sim/humanoid_gym/play_h5.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+    # Create dataset for actions
+    max_timesteps = stop_state_log
+    num_dof = env.num_dof
+    dset_actions = h5_file.create_dataset("actions", (max_timesteps, num_dof), dtype=np.float32)
+
+    # Create dataset of observations
+    buf_len = len(env.obs_history)  # length of observation buffer
+    dset_2D_command = h5_file.create_dataset("observations/2D_command", (max_timesteps, buf_len, 2), dtype=np.float32)
+    dset_3D_command = h5_file.create_dataset("observations/3D_command", (max_timesteps, buf_len, 3), dtype=np.float32)
```
Comment (WT-MM) [inline sim/humanoid_gym/play_h5.py:89]: explain what these are

File: sim/humanoid_gym/play_h5.py [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        )
+
+        fourcc = cv2.VideoWriter_fourcc(*"MJPG")
+
+        # Creates a directory to store videos.
+        video_dir = run_dir() / "videos"
+        experiment_dir = video_dir / train_cfg.runner.experiment_name
+        experiment_dir.mkdir(parents=True, exist_ok=True)
+
+        dir = os.path.join(experiment_dir, datetime.now().strftime("%b%d_%H-%M-%S") + str(args.run_name) + ".mp4")
```
Comment (WT-MM) [inline sim/humanoid_gym/play_h5.py:120]: Make the video and h5 file names the same

File: sim/humanoid_gym/play_h5.py (hunk 1 above)
Comment (WT-MM) [reply sim/humanoid_gym/play_h5.py:79]: https://github.com/kscalelabs/sim/pull/50#discussion_r1712163496

File: sim/humanoid_gym/play_h5.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
+import argparse
```
Comment (budzianowski) [inline sim/humanoid_gym/play_h5.py:1]: Please create a separate PR for model an keep only the data logic here.

**Review by budzianowski — CHANGES_REQUESTED** [review]

File: sim/humanoid_gym/play_h5.py (hunk 6 above)
Comment (wuallenwu) [reply sim/humanoid_gym/play_h5.py:1]: https://github.com/kscalelabs/sim/issues/51

i made one here, i'll clean up this pr

File: sim/humanoid_gym/envs/stompy_mini_env.py (hunk 2 above)
Comment (wuallenwu) [reply sim/humanoid_gym/envs/stompy_mini_env.py:268]: addressing in new pr

File: sim/humanoid_gym/play_h5.py (hunk 3 above)
Comment (wuallenwu) [reply sim/humanoid_gym/play_h5.py:171]: fixed!

File: sim/humanoid_gym/play_h5.py (hunk 4 above)
Comment (wuallenwu) [reply sim/humanoid_gym/play_h5.py:89]: fixed!

File: sim/humanoid_gym/play_h5.py (hunk 5 above)
Comment (wuallenwu) [reply sim/humanoid_gym/play_h5.py:120]: fixed

**Comment by budzianowski** [comment]
This PR should only have h5 related things

**Comment by wuallenwu** [comment]
addressed changes r.e. putting play_h5 logic in play

**Review by budzianowski — APPROVED** [review]

**Pull request merged**